cmd/zpool/zpool_iter.c

```diff
@@ -521,30 +521,80 @@ vdev_run_cmd(vdev_cmd_data_t *data, char *cmd)
 			free(env[i]);
 }
 
+/* run our command based on it's full path */
+static int
+vdev_run_cmd_fullpath(char *fullpath, vdev_cmd_data_t *data)
+{
+	if (fullpath == NULL || fullpath[0] == '\0')
+		return (1);
+
+	/* Does the script exist in path? */
+	if (access(fullpath, X_OK) == 0) {
+		vdev_run_cmd(data, fullpath);
+		return (0);
+	}
+
+	return (1);
+}
+
+/* Run our command script */
+static int
+vdev_run_cmd_path(char *path, char *cmd, vdev_cmd_data_t *data)
+{
+	char fullpath[MAXPATHLEN];
+
+	if (path == NULL || path[0] == '\0')
+		return (1);
+
+	if (cmd == NULL || cmd[0] == '\0')
+		return (1);
+
+	if (snprintf(fullpath, sizeof (fullpath), "%s/%s", path, cmd) == -1)
+		return (1);
+
+	return (vdev_run_cmd_fullpath(fullpath, data));
+}
+
 /* Thread function run for each vdev */
 static void
 vdev_run_cmd_thread(void *cb_cmd_data)
 {
 	vdev_cmd_data_t *data = cb_cmd_data;
 	const char *sep = ",";
 	char *cmd = NULL, *cmddup, *rest;
-	char fullpath[MAXPATHLEN];
+	char *userdir = NULL;
+	const char *hdir;
 
 	cmddup = strdup(data->cmd);
 	if (cmddup == NULL)
 		return;
 
+	hdir = getenv("HOME");
+	if (hdir != NULL) {
+		if (asprintf(&userdir, "%s/.zfs/zpool", hdir) == -1) {
+			userdir = NULL;
+		}
+	}
+
 	rest = cmddup;
 	while ((cmd = strtok_r(rest, sep, &rest))) {
-		if (snprintf(fullpath, sizeof (fullpath), "%s/%s",
-		    ZPOOL_SCRIPTS_DIR, cmd) == -1)
+		/* if cmd starts with '/', assume full path */
+		if (cmd[0] == '/') {
+			vdev_run_cmd_fullpath(cmd, data);
 			continue;
+		}
 
-		/* Does the script exist in our zpool scripts dir? */
-		if (access(fullpath, X_OK) == 0)
-			vdev_run_cmd(data, fullpath);
+		/* attempt to run the cmd from the user's dir first */
```

Comment:
You'll want to suppress the `Cant open /root/.zfs/zpool scripts dir` message since most people won't have that directory created:
```
./cmd/zpool/zpool status -c
Can't open /root/.zfs/zpool scripts dir
Current scripts in /usr/local/etc/zfs/zpool.d:
  upath           Show the underlying path for a device.
  smart           Show SMART temperature and important error stats (specific to drive type)
...
```